phoenix-tools/src/main/java/org/apache/phoenix/schema/SchemaSynthesisProcessor.java

```diff
@@ -0,0 +1,261 @@
+        List<ColumnDef> oldColumnDef = createStmt.getColumnDefs();
+        List<ColumnDef> newColumnDef = new ArrayList<>();
+        newColumnDef.addAll(oldColumnDef);
+        DropColumnStatement dropStmt = alterStatement;
+        for(ColumnName cName : dropStmt.getColumnRefs()) {
+            for(ColumnDef colDef : oldColumnDef) {
+                if(colDef.getColumnDefName().equals(cName)) {
+                    newColumnDef.remove(colDef);
+                    break;
+                }
+            }
+        }
+        newCreateStmt = new CreateTableStatement(createStmt, newColumnDef);
+        return newCreateStmt;
+    }
+
+    private CreateTableStatement getCreateTableStatement(AddColumnStatement alterStatement,
+            CreateTableStatement createStmt) throws Exception {
+        CreateTableStatement newCreateStmt = null;
+        String tableName = createStmt.getTableName().toString();
+        String tableNameInAlter = alterStatement.getTable().toString().trim();
+        sanityCheck(tableName, tableNameInAlter);
+        AddColumnStatement addStmt = alterStatement;
+        List<ColumnDef> oldColDef = createStmt.getColumnDefs();
+        List<ColumnDef> newColDef = new ArrayList<>();
+        if (addStmt.getColumnDefs() == null) {
+            ListMultimap<String, Pair<String, Object>>
+                    finalProps = getEffectiveProperties(addStmt, createStmt.getProps());
+            newCreateStmt = new CreateTableStatement(createStmt, finalProps, oldColDef);
+        } else {
+            newColDef.addAll(oldColDef);
+            newColDef.addAll(addStmt.getColumnDefs());
+            newCreateStmt = new CreateTableStatement(createStmt, newColDef);
+        }
+        return newCreateStmt;
+    }
+
+    private void sanityCheck(String tableName, String tableNameInAlter) throws Exception {
+        if (!tableName.equalsIgnoreCase(tableNameInAlter)) {
+            throw new Exception(ENTITY_NAME_IN_BASE_AND_ALTER_DDL_DON_T_MATCH);
+        }
+    }
+
+    private ListMultimap<String, Pair<String, Object>> getEffectiveProperties(
+            AddColumnStatement addStmt, ListMultimap<String, Pair<String, Object>> oldProps) {
+        Map<String, Object> oldPropMap = new HashMap();
+        Map<String, Object> changePropMap = new HashMap();
+
+        for (Pair<String, Object> value : oldProps.values()) {
+            oldPropMap.put(value.getFirst(),value.getSecond());
+        }
+        for (Pair<String, Object> value : addStmt.getProps().values()) {
+            changePropMap.put(value.getFirst(),value.getSecond());
+        }
+
+        oldPropMap.putAll(changePropMap);
+        ListMultimap<String, Pair<String, Object>>
+                finalProps =
+                ArrayListMultimap.<String, Pair<String, Object>>create();
+        for (Map.Entry<String, Object> entry : oldPropMap.entrySet()) {
+            finalProps.put("", Pair.newPair(entry.getKey(), entry.getValue()));
```

Comment:
What's the purpose of a hard-coded ["" -> [key -> value]] instead of just [key -> value]? Can't be duplicates, because oldPropMap was already a multi-map, right?

Reply:
Property multimap is formed with String and Pair as a key-value. Key for all properties pair is same "" but the value(pair of property key, value) is different. I had to stick to this to create a new CreateTableStatement. Not sure if I misunderstood your suggestion. LMK.